src/java.base/share/classes/java/util/zip/ZipOutputStream.java

```diff
@@ -205,9 +210,15 @@ public void putNextEntry(ZipEntry e) throws IOException {
         case DEFLATED:
             // store size, compressed size, and crc-32 in data descriptor
             // immediately following the compressed entry data
-            if (e.size  == -1 || e.csize == -1 || e.crc   == -1)
+            if (e.size  == -1 || e.csize == -1 || e.crc   == -1) {
                 e.flag = 8;
-
+            }
+            else if (!e.manual_csize) {
```

Comment:
I assume the existing expression that set if csize has been set so that we don't set the flag to 8 in two branches.

Reply:
I was thinking about that myself